### PR TITLE
Support replica set

### PR DESCRIPTION
**Pull request by paton**
Apologies for the messy PR and lack of documentation. Hopefully will find time to clean this up soon (or if anyone wants to adapt this in to a better PR, definitely supportive of that!)

### CODE REVIEW

**Comment by wulfsolter** [comment]
Nice work @paton 
Thank you

File: db.js [hunk 1]
```diff
@@ -14,8 +14,17 @@ var connect = function(config) {
...
   };
-  var db = new mongodb.Db('local', new mongodb.Server(host, port, dbOptions), {safe:true});
 
+  var db;
+
+  if (Array.isArray(host)) {
+    host = host.map(function(host) {
+      return new mongodb.Server(host, port, dbOptions);
+    });
+    db = new mongodb.Db('local', new mongodb.ReplSet(host), { safe: true, w: 0 });
```
Comment (nodesocket) [inline db.js]: Don't you need to pass in the replica set name?

**Comment by nodesocket** [comment]
Also, can we get documentation on how the `config.js` file looks using a replica set?

**Comment by paton** [comment]
@nodesocket Regarding replica set name, I originally tried passing it in but was getting some errors. Removing it fixed that issue, but from what I've seen, the standard practice is to pass it in.

Re: docs, simply pass in an array of hosts to the "server" option instead of a single host. For example

mongodb: {
    server: ['mongoserver1.com', 'mongoserver2.com', 'mongoserver3.com'],
    port: 27000,
    ...
}

Unfortunately I'm jammed right now and don't have time to make these improvements, but if I (or someone) has time to refactor this, a much better approach than this PR would be to allow a mongodb connection string instead of passing in hostname, ports, RS name, etc. http://docs.mongodb.org/manual/reference/connection-string/

All of those variables are embedded in a mongo connection string.

**Comment by ceefour** [comment]
@paton How do I do this from Docker image's ? `ME_CONFIG_MONGODB_SERVER` environment variable doesn't seem to support array.....

**Comment by paton** [comment]
Hm, I'm not familiar enough with Docker or how env vars are handled in this repo.

@wulfsolter might know more.

**Comment by knickers** [comment]
The problem will be with Linux environment variables. They don't support arrays, or at least I have never figured out how to do it.

**Comment by ceefour** [comment]
@knickers we can use commas. or better yet just use connection string

**Pull request merged**